packages/core-v2-sdk/src/utils/token/encodeTokenId.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Get a token id
+ * @param {number} maturity - The maturity timestamp in milliseconds
+ * @param {number} startTime - The optional start time timestamp in milliseconds
+ * @param {boolean} isYieldToken - Whether or not the token is a yield token
+ * @return A token id
+ */
+export function encodeTokenId(
+  maturity: number,
+  startTime = 0,
+  isYieldToken = false,
+): string {
+  const firstBit = isYieldToken ? 8 : 0;
+  const startTimeBits = (startTime / 1000).toString(16).padStart(31, "0");
+  const maturityHex = (maturity / 1000).toString(16);
+  const maturityBits = maturityHex.padStart(32, "0");
+
+  return isYieldToken || startTime
```

Comment:
Both of these values serve the same boolean purpose. If a start time exist then the token is a yield token. Otherwise if we use the function like this `encodeTokenid(101, 90, false` it generates a token id that is not valid in the protocol.

We should just use the start time value as the source of truth if we are generating a yt or not